servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1478,8 +1517,180 @@ public void setState(long fileId, SetStateOptions options)
     }
   }
 
-  private void setStateInternal(long fileId, long opTimeMs, SetStateOptions options)
+  /**
+   * Schedules a file for async persistence.
+   *
+   * @param fileId the id of the file for persistence
+   * @return the id of the worker that persistence is scheduled on
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public long scheduleAsyncPersistence(long fileId) throws FileDoesNotExistException {
+    // find the worker
+    long workerId = getWorkerStoringFile(fileId);
+
+    if (workerId == IdUtils.INVALID_WORKER_ID) {
+      LOG.warn("No worker found to schedule async persistence for file {}", fileId);
+      // no worker found, do nothing
+      return workerId;
+    }
+
+    // update the state
+    synchronized (mInodeTree) {
```

Comment:
hmm, I think the journaling was removed in this call when I merged the `SCHEDULED` and `PERSISTING` state into `IN_PROGRESS`. But now I think it was a mistake because of tolerance issue. It's possible master crashes between an async persist is scheduled and the worker heartbeat picks it up.

Will change it. Thanks for catching this issue.